### PR TITLE
Add scroll position saving

### PR DESCRIPTION
**Pull request by tkiyohar**
## Summary

-- Reimplementation of previous PR that was wrongly repealed: "Add scroll position saving #1935" --
-- Additionally includes some code changes and improvements to the original scroll position saving code --
Adds scroll position saving to the Home and Notifications app screens, fixes resizing error with home screen web cards (needed for scroll position saving), and adds loading screens to home screen web cards (prevents content flickering + looks much nice with scroll position saving).
## Changelog

[General] [Add] - Adds scroll position saving to home and notifications pages (also includes code from Gradle update PR: "Update Gradle Version and Settings #2017")

## Test Plan

Test the home page and notifications page for any bugs and to see if their scroll position is maintained when switching between screens using the bottom nav bar after messing around on either screen (e.g., hiding or unhiding cards on the home screen from either the home screen itself or the profile screen, toggling "I'm going" on certain free food events, etc). 

Also test logging in and out of student, guest and employee accounts to make sure the scroll position on both the home and notifications pages are reset when logging in/out of an account.

If this PR is successful, the scroll position should be seamlessly saved on the notifications and home screens without any jumpy behavior when reloading these screens. Additionally, web cards (e.g., "Campus Information" and "Health & Well-being" cards) should now display a loading screen while their contents are loading in. No other behavior of the app should be affected.

### CODE REVIEW

File: lib/core/providers/user.dart [hunk 1]
```diff
@@ -16,6 +17,8 @@ import 'package:pointycastle/asymmetric/api.dart';
 import 'package:pointycastle/asymmetric/oaep.dart';
 import 'package:pointycastle/pointycastle.dart' as pc;
 
+import '../../ui/home/home.dart';
```
Comment (c3bryant) [inline lib/core/providers/user.dart:21]: TODO (later): Refactor scroll controller reset methods to campus mobile core

File: lib/core/providers/cards.dart [hunk 2]
```diff
@@ -335,6 +336,10 @@ class CardsDataProvider extends ChangeNotifier {
   }
 
   void toggleCard(String card) {
+    debugPrint("THIS IS THE CURRENT CARD BEING TOGGLED: " + card);
+    if ((card == "student_health_wellbeing" || card == "student_info" || card == "campus_info") && _cardStates![card]!) {
```
Comment (c3bryant) [inline lib/core/providers/cards.dart]: Explore dynamically applying resetCardHeight

**Review by c3bryant — DISMISSED** [review]
Explore dynamically applying resetCardHeight in cards provider

**Review by morebytes — APPROVED** [review]
Looks good, fully tested

**Pull request merged**